sklearn/metrics/_classification.py

```diff
@@ -143,6 +145,32 @@ def _weighted_sum(sample_score, sample_weight, normalize=False):
         return sample_score.sum()
 
 
+def _nan_average(scores, weights):
```

Comment:
I wonder if we should not be calling this one `nanaverage` and adding it in `sklearn.utils.extmath.py`. Any thoughts @thomasjpfan?

Reply:
I'm okay with including `nanaverage` into `skelarn.utils.extmath`. Although I'll make it private by calling it `_nanaverage`.